tests/backup/backup_test.go

```diff
@@ -5705,7 +5736,264 @@ var _ = Describe("{ViewOnlyFullBackupRestoreIncrementalBackup}", func() {
 		log.Infof("Cleaning cloud accounts")
 		DeleteCloudAccounts(backupLocationMap, credName, cloudCredUID, ctx)
 	})
+})
+
+var _ = Describe("{IssueMultipleRestoresWithNamespaceAndStorageClassMapping}", func() {
+	var (
+		contexts          []*scheduler.Context
+		appContexts       []*scheduler.Context
+		bkpNamespaces     []string
+		clusterUid        string
+		clusterStatus     api.ClusterInfo_StatusInfo_Status
+		backupLocationUID string
+		cloudCredName     string
+		cloudCredUID      string
+		bkpLocationName   string
+		backupName        string
+		userName          string
+		userCtx           context.Context
+		scName            string
+		restoreList       []string
+		sourceScName      *storageapi.StorageClass
+	)
+	timestamp := time.Now().Unix()
+	namespaceMap := make(map[string]string)
+	backupLocationMap := make(map[string]string)
+	labelSelectors := make(map[string]string)
+	bkpNamespaces = make([]string, 0)
+	storageClassMapping := make(map[string]string)
+	k8sStorage := storage.Instance()
+	params := make(map[string]string)
+
+	JustBeforeEach(func() {
+		StartTorpedoTest("IssueMultipleRestoresWithNamespaceAndStorageClassMapping",
+			"Issue multiple restores with namespace and storage class mapping", nil, 82945)
+		log.InfoD("Deploy applications needed for backup")
+		contexts = make([]*scheduler.Context, 0)
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			taskName := fmt.Sprintf("%s-%d", taskNamePrefix, i)
+			appContexts = ScheduleApplications(taskName)
+			contexts = append(contexts, appContexts...)
+			for _, ctx := range appContexts {
+				ctx.ReadinessTimeout = appReadinessTimeout
+				namespace := GetAppNamespace(ctx, taskName)
+				bkpNamespaces = append(bkpNamespaces, namespace)
+			}
+		}
+	})
+	It("Issue multiple restores with namespace and storage class mapping", func() {
+		namespaceMap[bkpNamespaces[0]] = fmt.Sprintf("new-namespace-%v", time.Now().Unix())
+		ctx, err := backup.GetAdminCtxFromSecret()
+		dash.VerifyFatal(err, nil, "Getting px-central-admin context")
+		providers := getProviders()
+		bucketNames := getBucketName()
+		Step("Validate applications", func() {
+			log.InfoD("Validate applications")
+			ValidateApplications(contexts)
+		})
 
+		Step("Create user", func() {
+			log.InfoD("Create user")
+			userName = fmt.Sprintf("testuser")
```

Comment:
I would suggest you append the username with the test case ID so that it is recognisable in keycloak.

Reply:
done